sol/table.hpp

```diff
@@ -98,9 +98,24 @@ class table : public reference {
         return *this;
     }
 
+    template<typename Fx>
+    void for_each(Fx&& fx) const {
+        push();
+        stack::push(state(), nil);
+        while (lua_next(this->state(), -2)) {
+            sol::object key(state(), -2);
+            sol::object value(state(), -1);
+            fx(key, value);
+            lua_pop(state(), 1);
+        }
+        pop();
+    }
+
     size_t size() const {
         push();
-        return lua_rawlen(state(), -1);
+        size_t result = lua_rawlen(state(), -1);
+        pop();

```

Comment:
It actually left a `table` instance on the stack, since `lua_rawlen` does not pop the table or push anything on the stack (lua_len does push something onto the stack).
See: http://puu.sh/hOU9T/9efa5d70e9.png